### PR TITLE
New Feesh Steal Logic

### PR DESCRIPTION
**Pull request by MrOutWeb**
I've developed a reworked version of the steal command for the feesh cog.
Notable features:
- Based Success/Fail/Drop on a success chance rather than a fail chance
- Swapped to using a tanh based function rather than an exponential one
- Standardised feesh stealing odds based on feesh top
- Increased Success chance at large differences, decreased Drop chance at large diffences
- Increased Drop chance at small differences, minimal change in Success chance at low differences
- Max steal chance ~ 40%. Min steal chance varies, currently ~10%

### CODE REVIEW

File: cogs/Feesh/feesh.py [hunk 1]
```diff
@@ -423,17 +423,25 @@ def isvalid(m):
...
-                    return 1
-                else:
-                    # x > 0
-                    # 1 to 0.5
-                    return (np.exp(-x / 2) + 1) / 2
-
-            fail = odds(feesh_diff)
-
-            drop = (1 - fail) / 2  # success = 1. - fail - drop
+			def odds(y):
```
Comment (Snaptraks) [inline cogs/Feesh/feesh.py]: You should be using a 4-spaces tab, otherwise the interpreter will complain

File: cogs/Feesh/feesh.py [hunk 2]
```diff
@@ -423,17 +423,25 @@ def isvalid(m):
...
+
+				L = np.array(L, dtype=[('ID', '<i8'), ('amount', '<i8'), ])
+				top_amount = L['amount'].max()
+
+				if y <= 0:
+					return 0
+				else:
+					# y > 0
+					# 0 to 0.4ish
+					return (np.tanh((y-(1.2*top))/top)/2+0.5)
```
Comment (Snaptraks) [inline cogs/Feesh/feesh.py]: `top` is undefined? Maybe you meant `top_amount`

Comment (MrOutWeb) [reply cogs/Feesh/feesh.py]: Yes, my apologies, missed that.

File: cogs/Feesh/feesh.py (hunk 1 above)
Comment (MrOutWeb) [reply cogs/Feesh/feesh.py]: Didn't realise, I think I was trying to not spam space by using tabs.

**Comment by MrOutWeb** [comment]
I can address both of these issues when I get a moment.

**Comment by MrOutWeb** [comment]
@Snaptraks I've addressed the issues you brought up.

**Pull request merged**